src/iuse.cpp

```diff
@@ -2747,160 +2747,572 @@ int iuse::sew(player *p, item *it, bool, point)
+    //minimum LL_LOW of LL_DARK + (ELFA_NV or atomic_light)
+    if (p->fine_detail_vision_mod() > 4) {
+        add_msg(m_info, _("You can't see to sew!"));
         return 0;
     }
+        int thread_used = 5;
 
-    int items_needed = (fix->damage > 2 || fix->damage == 0) ? 1 : 0;
-
-    // this will cause issues if/when NPCs start being able to sew.
-    // but, then again, it'll cause issues when they start crafting, too.
-    const inventory &crafting_inv = p->crafting_inventory();
-    bool bFound = false;
-    //go through all discovered repair items and see if we have any of them available
-    for( auto &repair_items_i : repair_items ) {
-        if( crafting_inv.has_amount( repair_items_i, items_needed ) ) {
-            //we've found enough of a material, use this one
-            repair_item = repair_items_i;
-            bFound = true;
-        }
-    }
-    if (!bFound) {
-        for (unsigned int i = 0; i < repair_items.size(); i++) {
-            p->add_msg_if_player(m_info, _("You don't have enough %s to do that."), plurals[i].c_str());
+        int pos = g->inv_for_filter( _("Enhance what?"), []( const item & itm ) {
+            return itm.made_of( "cotton" ) ||
+                   itm.made_of( "leather" ) ||
+                   itm.made_of( "fur" ) ||
+                   itm.made_of( "nomex" ) ||
+                   itm.made_of( "plastic" ) ||
+                   itm.made_of( "wool" );
+        } );
+        item *mod = &(p->i_at(pos));
+        if (mod == NULL || mod->is_null()) {
+            p->add_msg_if_player(m_info, _("You do not have that item!"));
+            return 0;
+        };
+                itype_id repair_item = "none";
+        std::vector<std::string> plurals;
+        std::vector<itype_id> repair_items;
+        std::string plural = "";
+        //translation note: add <plural> tag to keep them unique
+        if (mod->made_of("cotton")) {
+            repair_items.push_back("rag");
+            plurals.push_back(rm_prefix(_("<plural>rags")));
+        }
+        if (mod->made_of("leather")) {
+            repair_items.push_back("leather");
+            plurals.push_back(rm_prefix(_("<plural>leather")));
+        }
+        if (mod->made_of("fur")) {
+            repair_items.push_back("fur");
+            plurals.push_back(rm_prefix(_("<plural>fur")));
+        }
+        if (mod->made_of("nomex")) {
+            repair_items.push_back("nomex");
+            plurals.push_back(rm_prefix(_("<plural>nomex")));
+        }
+        if (mod->made_of("plastic")) {
+            repair_items.push_back("plastic");
+            plurals.push_back(rm_prefix(_("<plural>plastic")));
+        }
+        if (mod->made_of("wool")) {
+            repair_items.push_back("felt_patch");
+            plurals.push_back(rm_prefix(_("<plural>wool")));
+        }
+        if (repair_items.empty()) {
+            p->add_msg_if_player(m_info, _("Your %s is not made of fabric, leather, fur, or wool."),
```

Comment:
Maybe add "or plastic" here as it can be used on plastic items as well. Not that it bothers me, but some player will nitpick this (-: